src/modules/registration/components/Registration.js

```diff
@@ -1,43 +1,146 @@
 
 export const Registration = () => {
+    const [firstName, setFirstName] = useState('')
+    const [lastName, setLastName] = useState('')
+    const [username, setUsername] = useState('')
+    const [email, setEmail] = useState('')
+    const [password, setPassword] = useState('')
+    const [error, setError] = useState(null)
+    const [isLoading, setIsLoading] = useState(false)
+    const [success, setSuccess] = useState(false)
+
+    const handleFirstName = (event) => {
+        setFirstName(event.target.value)
+    }
+
+    const handleLastName = (event) => {
+        setLastName(event.target.value)
+    }
+
+    const handleUsername = (event) => {
+        setUsername(event.target.value)
+    }
+
+    const handleEmail = (event) => {
+        setEmail(event.target.value)
+    }
+
+    const handlePassword = (event) => {
+        setPassword(event.target.value)
+    }
+
+    const handleSubmit = async (event) => {
+        event.preventDefault()
+        setIsLoading(true)
+        try {
+            const response = await axios.post(
+                'https://cf4lq5bf3denzvriyvqnuxlnxu0gwtgo.lambda-url.ap-southeast-2.on.aws/',
+                {
+                    firstName,
+                    lastName,
+                    username,
+                    email,
+                    password,
+                },
+                {
+                    headers: {
+                        'Access-Control-Allow-Origin': '*',
+                    },
+                    mode: 'cors',
+                }
+            )
+            console.log(response.data)
+            setSuccess(true)
+        } catch (error) {
+            console.log('Error:', error)
+            setError(error)
+        } finally {
+            setIsLoading(false)
+        }
+    }
+
     return (
-        <div className={styles.container}>
-            <input
-                className={styles.input}
-                type="text"
-                placeholder="First name"
-                required
-            />
-
-            <input
-                className={styles.input}
-                type="text"
-                placeholder="Last name"
-                required
-            />
-
-            <input
-                className={styles.input}
-                type="text"
-                placeholder="Email"
-                required
-            />
-
-            <input
-                className={styles.input}
-                type="password"
-                placeholder="Password"
-                required
-            />
-
-            {/* Insert RegisterButton component here */}
-            <button className={styles.registerButton}>Register</button>
-            {/*Insert Login route here*/}
-            <button className={styles.returnToLoginButton}>
-                Already have an account?
-            </button>
-        </div>
+        <>
+            {success ? (
+                /* success, need to redirect to a user logged in homepage */
```

Comment:
You should be able to do this already. 
Let me know if you need an hand

Reply:
Warning: Cannot update a component (`RouterProvider`) while rendering a different component (`Registration`)